ruby/engine/embedded_help.rb

```diff
@@ -56,6 +58,17 @@ def self.preprocess_ruby_script(s)
 # DLM: ignore for now
 #Encoding.default_external = Encoding::ASCII
 
+# We patch Kernel.open, and IO.open
+# to read embedded files as a StringIO, not as a FileIO
+# But Gem.open_file for eg expects it to be a File, not a StringIO, and on
+# Windows it will call File::flock (file lock) to protect access, but StringIO
+# does not have this method
+class FakeFileAsStringIO < StringIO
+  def flock
+    return 0
+  end
+end
```

Comment:
In Kernel.open and IO.open, when we returned a StringIO, we now return this FakeFileAsStringIO which will have the flock method...